gcp/website/utils.py

```diff
@@ -15,13 +15,34 @@
 
 import os
 from datetime import datetime, UTC
+from functools import cache
 
+import google.auth
+from google.auth.exceptions import DefaultCredentialsError
 
-def is_prod():
-  # TODO(michaelkedar):
-  # This function actually checks if it's running on gcp (on prod OR staging)
-  # and it's only used for Redis cache (which has its own env vars) and logging.
-  # Consider removing this altogether.
+
+@cache  # google.auth.default() can be a bit expensive.
+def api_url() -> str:
+  """
+  Returns the URL of the OSV API for the current GCP project.
+
+  Falls back to 'api.osv.dev' if URL cannot be determined.
+  """
+  try:
+    _, project = google.auth.default()
+  except DefaultCredentialsError:
+    project = '<unknown>'
+
+  # TODO(michaelkedar): Is there a way to avoid hard-coding this?
```

Comment:
Might it be easier (and potentially safer) to hard-code the production project ID (which is somewhat inescapable, and also more durable) and fail open to the non-production hostname for everything else?

Reply:
Changed it to fall back to `api.test.osv.dev`.